### PR TITLE
Implement preprocessor directive ifdef-else-endif

### PR DESCRIPTION
**Pull request by ChinYikMing**
#43

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Append `Close #43` at the end of git commit message.

**Comment by jserv** [comment]
Check [CONTRIBUTING.md](https://github.com/jserv/shecc/blob/master/CONTRIBUTING.md) to enforce consistent style.

**Comment by ChinYikMing** [comment]
passed the checks

File: tests/driver.sh [hunk 1]
```diff
@@ -323,4 +323,20 @@ int main()
...
+#define B 200
+int main()
+{
+    int x;
+#ifdef A
+    x = A;
+#else
+    x = B;
+#endif
+    return x;
```
Comment (jserv) [inline tests/driver.sh:338]: The return value is crucial for automated tests. Ensure that 0 is returned in the expected path.

**Review by jserv — CHANGES_REQUESTED** [review]
The string `Close #43` should appear in the body of git commit message rather than the subject. In addition, read https://cbea.ms/git-commit/ carefully to consolidate the commit message.

**Comment by jserv** [comment]
I defer to @eecheng87 to validate the proposed changes.

**Review by eecheng87 — APPROVED** [review]

**Comment by jserv** [comment]
Thank @ChinYikMing for contributing!
You may consider to implement `#if defined(...)` and `#elif`, so that we can refine the existing code such as [lib/c](https://github.com/jserv/shecc/blob/master/lib/c.c). Expected usage:
```c
#if defined(__arm__)
#define __syscall_exit 1
#define __syscall_read 3
#define __syscall_write 4
#define __syscall_close 6
#define __syscall_open 5
#define __syscall_brk 45

#elif defined(__riscv)
#define __syscall_exit 93
#define __syscall_read 63
#define __syscall_write 64
#define __syscall_close 57
#define __syscall_open 1024
#define __syscall_openat 56
#define __syscall_brk 214

#else
#error Unsupported architecture
#endif
```

**Pull request merged**